app/services/preserved_object_handler.rb

```diff
@@ -343,12 +343,25 @@ def with_active_record_transaction_and_rescue
 
   # expects @incoming_version to be numeric
   def version_comparison_results(db_object, version_symbol)
+    db_version =
+      case db_object.class
+      when PreservedCopy
+        db_object.version
+      when PreservedObject
+        db_object.current_version
+      end
```

Comment:
i think instead you could just extract the result of `db_object.send(version_symbol)` into a variable, right?  would probably help make the conditional below more readable too, since it appears to be asking for that same piece of info in each of its branches.

alternatively, we could keep the above change and use `db_version` in place of the `db_object.send(version_symbol)`, and get rid of the `version_symbol` param, which always did feel a little icky from an object oriented design perspective.  i might actually be in favor of that.

Reply:
Good 👀 - I somehow missed that whole :version_symbol thing.   I added a commit that changes the argument passed to the actual db_version, which cleans everything up.